### ISSUE TITLE
[match] force-generated profile does not include all capabilities

### ISSUE THREAD

**Issue by ojii** [fastlane/fastlane#15834]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I have an existing app for which I use `fastlane` & `match` to handle provisioning profiles. After adding the "App Groups" capability/entitlement in XCode and running `fastlane match --force -a appids,...` again, even though it reported to have successfully re-created the profiles, the new profile does not include the "App Groups" capability.

##### Command executed
<details><summary>fastlane match development --force -a appid,appid.watchkitapp,appid.watchkitapp.watchkitextension</summary>

```
$ fastlane match development --force -a appid,appid.watchkitapp,appid.watchkitapp.watchkitextension 
[✔] 🚀 
[14:29:21]: Successfully loaded '.../fastlane/Matchfile' 📄

+---------+-------------------------------------------------+
|        Detected Values from './fastlane/Matchfile'        |
+---------+-------------------------------------------------+
| git_url | git@github.com:org/certs.git |
+---------+-------------------------------------------------+


+----------------------------+-----------------------------------------------------------------------------------------------------------+
|                                                       Summary for match 2.138.0                                                        |
+----------------------------+-----------------------------------------------------------------------------------------------------------+
| force                      | true                                                                                                      |
| app_identifier             | ["appid", "appid.watchkitapp", "appid.watchkitapp.watchkitextension"] |
| type                       | development                                                                                               |
| readonly                   | false                                                                                                     |
| generate_apple_certs       | true                                                                                                      |
| skip_provisioning_profiles | false                                                                                                     |
| username                   | appleid@email.tld                                                                                  |
| storage_mode               | git                                                                                                       |
| git_url                    | git@github.com:org/certs.git                                                           |
| git_branch                 | master                                                                                                    |
| shallow_clone              | false                                                                                                     |
| clone_branch_directly      | false                                                                                                     |
| keychain_name              | login.keychain                                                                                            |
| force_for_new_devices      | false                                                                                                     |
| skip_confirmation          | false                                                                                                     |
| skip_docs                  | false                                                                                                     |
| platform                   | ios                                                                                                       |
| verbose                    | false                                                                                                     |
+----------------------------+-----------------------------------------------------------------------------------------------------------+

[14:29:21]: Cloning remote git repo...
[14:29:21]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[14:29:26]: 🔓  Successfully decrypted certificates repo
[14:29:26]: Verifying that the certificate and profile are still valid on the Dev Portal...
Available session is not valid any more. Continuing with normal login.
[14:29:34]: Installing certificate...

+-------------------+------------------------------------------------+
|                       Installed Certificate                        |
+-------------------+------------------------------------------------+
| User ID           | **********                                     |
| Common Name       | Apple Development: ********** |
| Organisation Unit | **********                                     |
| Organisation      | **********                                    |
| Country           | US                                             |
| Start Datetime    | 2019-10-28 07:07:33 UTC                        |
| End Datetime      | 2020-10-27 07:07:33 UTC                        |
+-------------------+------------------------------------------------+


+-------------------------------------+-------------------------------------+
|                         Summary for sigh 2.138.0                          |
+-------------------------------------+-------------------------------------+
| app_identifier                      | appid                   |
| username                            | appleid@email.tld            |
| force                               | true                                |
| cert_id                             | **********                          |
| provisioning_name                   | match Development appid |
| ignore_profiles_with_different_name | true                                |
| platform                            | ios                                 |
| development                         | true                                |
| adhoc                               | false                               |
| developer_id                        | false                               |
| skip_install                        | false                               |
| skip_fetch_profiles                 | false                               |
| skip_certificate_verification       | false                               |
| readonly                            | false                               |
+-------------------------------------+-------------------------------------+

[14:29:35]: Starting login with user 'appleid@email.tld'
Available session is not valid any more. Continuing with normal login.
[14:29:38]: Successfully logged in
[14:29:38]: Fetching profiles...
[14:29:41]: Verifying certificates...
[14:29:44]: Found 1 matching profile(s)
[14:29:44]: Recreating the profile
[14:29:47]: Creating new provisioning profile for 'appid' with name 'match Development appid' for 'ios' platform
[14:29:48]: Downloading provisioning profile...
[14:29:49]: Successfully downloaded provisioning profile...
[14:29:49]: Installing provisioning profile...
/var/folders/rk/_qysk9hs40qcq14h44l57wch0000gn/T/d20191226-94118-16r217t/profiles/development/Development_appid.mobileprovision
[14:29:49]: Installing provisioning profile...

+-------------------------------------+-------------------------------------------------+
|                               Summary for sigh 2.138.0                                |
+-------------------------------------+-------------------------------------------------+
| app_identifier                      | appid.watchkitapp                   |
| username                            | appleid@email.tld                        |
| force                               | true                                            |
| cert_id                             | **********                                      |
| provisioning_name                   | match Development appid.watchkitapp |
| ignore_profiles_with_different_name | true                                            |
| platform                            | ios                                             |
| development                         | true                                            |
| adhoc                               | false                                           |
| developer_id                        | false                                           |
| skip_install                        | false                                           |
| skip_fetch_profiles                 | false                                           |
| skip_certificate_verification       | false                                           |
| readonly                            | false                                           |
+-------------------------------------+-------------------------------------------------+

[14:29:50]: Starting login with user 'appleid@email.tld'
Available session is not valid any more. Continuing with normal login.
[14:29:53]: Successfully logged in
[14:29:53]: Fetching profiles...
[14:29:54]: Verifying certificates...
[14:29:54]: Found 1 matching profile(s)
[14:29:54]: Recreating the profile
[14:29:57]: Creating new provisioning profile for 'appid.watchkitapp' with name 'match Development appid.watchkitapp' for 'ios' platform
[14:29:59]: Downloading provisioning profile...
[14:29:59]: Successfully downloaded provisioning profile...
[14:29:59]: Installing provisioning profile...
/var/folders/rk/_qysk9hs40qcq14h44l57wch0000gn/T/d20191226-94118-16r217t/profiles/development/Development_appid.watchkitapp.mobileprovision
[14:29:59]: Installing provisioning profile...

+-------------------------------------+-------------------------------------------------------------------+
|                                        Summary for sigh 2.138.0                                         |
+-------------------------------------+-------------------------------------------------------------------+
| app_identifier                      | appid.watchkitapp.watchkitextension                   |
| username                            | appleid@email.tld                                          |
| force                               | true                                                              |
| cert_id                             | **********                                                        |
| provisioning_name                   | match Development appid.watchkitapp.watchkitextension |
| ignore_profiles_with_different_name | true                                                              |
| platform                            | ios                                                               |
| development                         | true                                                              |
| adhoc                               | false                                                             |
| developer_id                        | false                                                             |
| skip_install                        | false                                                             |
| skip_fetch_profiles                 | false                                                             |
| skip_certificate_verification       | false                                                             |
| readonly                            | false                                                             |
+-------------------------------------+-------------------------------------------------------------------+

[14:30:00]: Starting login with user 'appleid@email.tld'
Available session is not valid any more. Continuing with normal login.
[14:30:02]: Successfully logged in
[14:30:02]: Fetching profiles...
[14:30:02]: Verifying certificates...
[14:30:03]: Found 1 matching profile(s)
[14:30:03]: Recreating the profile
[14:30:05]: Creating new provisioning profile for 'appid.watchkitapp.watchkitextension' with name 'match Development appid.watchkitapp.watchkitextension' for 'ios' platform
[14:30:06]: Downloading provisioning profile...
[14:30:06]: Successfully downloaded provisioning profile...
[14:30:06]: Installing provisioning profile...
/var/folders/rk/_qysk9hs40qcq14h44l57wch0000gn/T/d20191226-94118-16r217t/profiles/development/Development_appid.watchkitapp.watchkitextension.mobileprovision
[14:30:06]: Installing provisioning profile...
[14:30:08]: 🔒  Successfully encrypted certificates repo
[14:30:08]: Pushing changes to remote git repo...
[14:30:12]: Finished uploading files to Git Repo [git@github.com:org/certs.git]

+---------------------+-------------------------------------------------+---------------------------------------------------------------+
|                                                    Installed Provisioning Profile                                                     |
+---------------------+-------------------------------------------------+---------------------------------------------------------------+
| Parameter           | Environment Variable                            | Value                                                         |
+---------------------+-------------------------------------------------+---------------------------------------------------------------+
| App Identifier      |                                                 | appid                                             |
| Type                |                                                 | development                                                   |
| Platform            |                                                 | ios                                                           |
| Profile UUID        | sigh_appid_development              | eafc9c59-4194-44bd-a26e-d097f6777677                          |
| Profile Name        | sigh_appid_development_profile-name | match Development appid                           |
| Profile Path        | sigh_appid_development_profile-path | **********         |
|                     |                                                 | ********** |
| Development Team ID | sigh_appid_development_team-id      | **********                                                    |
+---------------------+-------------------------------------------------+---------------------------------------------------------------+


+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
|                                                          Installed Provisioning Profile                                                           |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
| Parameter           | Environment Variable                                        | Value                                                         |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
| App Identifier      |                                                             | appid.watchkitapp                                 |
| Type                |                                                             | development                                                   |
| Platform            |                                                             | ios                                                           |
| Profile UUID        | sigh_appid.watchkitapp_development              | 4d7988a1-5fc0-4ab6-a4c7-026742924abe                          |
| Profile Name        | sigh_appid.watchkitapp_development_profile-name | match Development appid.watchkitapp               |
| Profile Path        | sigh_appid.watchkitapp_development_profile-path | **********         |
|                     |                                                             | ********** |
| Development Team ID | sigh_appid.watchkitapp_development_team-id      | **********                                                    |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+


+---------------------+-----------------------------------------------------------------------------+-------------------------------------------------------------------+
|                                                                    Installed Provisioning Profile                                                                     |
+---------------------+-----------------------------------------------------------------------------+-------------------------------------------------------------------+
| Parameter           | Environment Variable                                                        | Value                                                             |
+---------------------+-----------------------------------------------------------------------------+-------------------------------------------------------------------+
| App Identifier      |                                                                             | appid.watchkitapp.watchkitextension                   |
| Type                |                                                                             | development                                                       |
| Platform            |                                                                             | ios                                                               |
| Profile UUID        | sigh_appid.watchkitapp.watchkitextension_development            | e65eca09-c500-4af3-8edc-24d393e6c446                              |
| Profile Name        | sigh_appid.watchkitapp.watchkitextension_development_profile-n  | match Development appid.watchkitapp.watchkitextension |
|                     | ame                                                                         |                                                                   |
| Profile Path        | sigh_appid.watchkitapp.watchkitextension_development_profile-p  | **********             |
|                     | ath                                                                         | **********     |
| Development Team ID | sigh_appid.watchkitapp.watchkitextension_development_team-id    | **********                                                        |
+---------------------+-----------------------------------------------------------------------------+-------------------------------------------------------------------+

[14:30:12]: All required keys, certificates and provisioning profiles are installed 🙌
```
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.1                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.24.1                                             |
| Installation Source         | ~/.gem/bin/fastlane                                            |
| Host                        | Mac OS X 10.15.1 (19B88)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.3                                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
before_all do
  setup_circle_ci
end

lane :ci do
  test(output_directory: "/Users/distiller/test_output/")
end

lane :ciUI do
  testUI(output_directory: "/Users/distiller/test_output/")
end

lane :test do |options|
  scheme = options[:scheme] ? options[:scheme] : "Tests"
  scan(scheme: scheme, device: "iPhone XS", code_coverage: true, skip_slack: true)
end

lane :testUI do |options|
  options[:scheme] = "UITests"
  test(options)
end

lane :profiles do |options|
  options[:type] = "appstore"
  run_match(options)
end

lane :dev_profiles do |options|
  options[:type] = "development"
  run_match(options)
end

private_lane :run_match do |options|
  options[:app_identifier] = ["appid", "appid.watchkitapp", "aooud.watchkitapp.watchkitextension"]
  match(options)
end

lane :beta do
  increment_build_number(
    build_number: latest_testflight_build_number + 1,
    xcodeproj: "my.xcodeproj",
  )
  sync_code_signing(
    type: "appstore",
    app_identifier: ["appid", "appid.watchkitapp", "appid.watchkitapp.watchkitextension"],
    readonly: true
  )
  build_app(scheme: "My Scheme", include_bitcode: true)
  upload_to_testflight
  slack(message: "Successfully distributed a new beta build")
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "appid"
apple_id "appleid@email.tld"
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.138.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| bundler                   | 2.1.2        |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.3        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.5        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.71.1       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.17.1       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.2       |
| multi_json                | 1.14.1       |
| os                        | 1.0.1        |
| signet                    | 0.12.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2019.1009  |
| mime-types                | 3.3          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.3.0        |
| google-cloud-core         | 1.4.1        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.3.0        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.7        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.7.0        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.2        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.14.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2019-12-26**
</details>

**Comment by ojii** [comment]
updated to `fastlane 2.140.0` but the issue still persists.

**Comment by joelrorseth** [comment]
I am experiencing a similar issue right now, `match` runs fine but does not create a profile with the App Groups capability.

`error: "OneSignalNotificationServiceExtension" requires a provisioning profile with the App Groups feature.`

**Comment by ojii** [comment]
fastlane 2.143.0, problem still happens.

**Comment by budde377** [comment]
Still happens on 2.145.0

**Comment by ojii** [comment]
still can't get this to work with `2.146.1`. does anyone know of a workaround (other than not using fastlane)?

**Comment by ikoryakin** [comment]
Still happens on 2.149.1

**Comment by ojii** [comment]
I found a workaround/solution for this:

I incorrectly assumed, fastlane would manage everything needed to get the certificates to work with appgroups, but apparently I was incorrect. To get fastlane to issue certificates with app groups correctly, you're gonna have to [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources/identifiers/list) in your Apple Developer Account, edit the Bundle ID for which you want to enable App Groups, select the App Groups capability and select the appropriate App Groups there. Once that is done, fastlane works as expected.
